core/controllers/admin.py

```diff
@@ -2261,6 +2262,29 @@ def put(self) -> None:
         self.render_json({})
 
 
+class RegenerateTopicSummariesHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Handler to regenerate the summaries of all topics."""
+
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {
+        'PUT': {}
+    }
+
+    @acl_decorators.can_access_admin_page
+    def put(self) -> None:
+        """Regenerates all topic summary models."""
+
+        topics = sorted(
```

Comment:
For the backend test covering this code, it expects that each topic ID to be passed into `generate_topic_summary`. However, by using any of the swap methods to make that assertion, it requires that the topic IDs to be passed as arguments in a *specific order*. On the other hand, `topic_fetchers.get_all_topics` in this code returns the topics in some *random order*. I didn't really know how to go about adding the additional testing infrastructure to assert arguments in any order, so I just went with sorting the topics instead.

Reply:
Done. I added a comment explaining this. I think I should also open an issue to allow the assertion of function calls based on counts and not their order.

Reply:
Thanks, sounds fine to me. 

Where do you need the functionality for asserting function call counts? If there's no immediate use case then I'd suggest not worrying about it for now and only bring it up when it's needed.

Reply:
So for our case, in `core/controllers/admin_test.py`:

```python
       with self.swap_with_checks(
                topic_services, 'generate_topic_summary',
                topic_services.generate_topic_summary,
                expected_args=[(topic_id,), (topic_id_2,)]):
            self.put_json(
                feconf.REGENERATE_TOPIC_SUMMARIES_URL, {},
                csrf_token=csrf_token, expected_status_int=200)
```

We expect `generate_topic_summary` to first be called with `topic_id` as an argument. Then, we expect `generate_topic_summary` to be called a second time with `topic_id_2` as an argument.

However, I want `generate_topic_summary` to be called twice, once with `topic_id` as an argument and the other with `topic_id_2` as an argument. I don't care whether `generate_topic_summary` was called first with `topic_id` and called second with `topic_id_2`, or if `generate_topic_summary` was called first with `topic_id_2` and called second with `topic_id`.

Reply:
Ah ok. Though if it's happening in a deterministic order I suppose it doesn't matter too much. I'm happy either way re filing the issue, though I lean slightly against since it doesn't seem that high-priority IMO.